command_line_assistant/daemon/database/manager.py

```diff
@@ -59,6 +60,9 @@ def _create_engine(self, echo: bool) -> Engine:
             connection_url = self._config.database.get_connection_url()
             # SQLite-specific settings
             if self._config.database.type == "sqlite":
+                if self._config.database.connection_string is not None:
+                    db_path = pathlib.Path(self._config.database.connection_string)
+                    db_path.parent.mkdir(parents=True, exist_ok=True)
```

Comment:
Can we use `utils.files.create_folder` for that? That could be used in production as well if the user does not change from sqlite to a more robust db. At least, we can avoid it being created with "wrong" permissions